### PR TITLE
[GOAL2-441] Implement default timeouts for REST Server 

### PR DESCRIPTION
**Pull request by EvanJRichard**
## Summary

Currently, there is no default timeout for handling a REST request. So, a caller could theoretically use a node's resources indefinitely by constructing some long-running request.

This PR proposes to add a default `WriteTimeout` and `ReadTimeout` to the API `http.Server`. More on what they do can be found in the `net.http` docs (or in summarizing articles, [like this one I used](https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/)). In short, at time of writing this PR proposes to add a ~30~ ~90~ 120 second request timeout.

## Test Plan

Automated tests are running now to check for regressions.

### CODE REVIEW

File: daemon/algod/server.go [hunk 1]
```diff
@@ -185,7 +186,12 @@ func (s *Server) Start() {
 	}
 
 	addr = listener.Addr().String()
-	server = http.Server{Addr: addr, Handler: handler}
+	server = http.Server{
+		Addr:         addr,
+		Handler:      apiHandler,
+		ReadTimeout:  15 * time.Second,
```
Comment (Karmastic) [inline daemon/algod/server.go]: Will this break the wait-for-block-after call that waits up to 1 minute?

Comment (EvanJRichard) [reply daemon/algod/server.go]: @Karmastic - Good find, yes, this timeout would trigger before the wait-for-block timeout. I'll raise the default timeout.

**Comment by EvanJRichard** [comment]
@tsachiherman wrote:
> Can you move these magic numbers into a constants or even better - into a config value ?
> These are the type of "tweaking values" that we might want to suggest in case one of our endpoints is performing poorly.
Good thinking, :+1: I'll implement that. 
> btw - 4 out of the 5 files included here have zero code changes. I don't see anything wrong with these changes.
Yeah, the pre-PR directions requested I run `make sanity`, those changes are the result of that.

File: config/local_defaults.go [hunk 2]
```diff
@@ -79,6 +79,8 @@ var defaultLocalV4 = Local{
 	OutgoingMessageFilterBucketSize:       128,
 	ReconnectTime:                         1 * time.Minute, // Was 60ns
 	ReservedFDs:                           256,
+	RestReadTimeoutSeconds:                15,
```
Comment (Karmastic) [inline config/local_defaults.go]: Thanks for alphabetizing them! :)

File: config/local_defaults.go [hunk 3]
```diff
@@ -286,6 +288,12 @@ func migrate(cfg Local) (newCfg Local, err error) {
 		if newCfg.PriorityPeers == nil {
 			newCfg.PriorityPeers = map[string]bool{}
 		}
+		if newCfg.RestReadTimeoutSeconds == defaultLocalV3.RestReadTimeoutSeconds {
```
Comment (Karmastic) [inline config/local_defaults.go]: Delete this block - should not be needed.

**Pull request merged**